src/components/Session/withAuthentication.js

```diff
@@ -13,7 +13,19 @@ const withAuthentication = Component => {
         authUser: null,
       };
     }
+    
+    componentDidMount() {
+      this.listener = this.props.firebase.auth.onAuthStateChanged(
```

Comment:
You should provide the function in the parameter. In this case the function will take authUser variable and that will be set to the authUser state if it is defined or authUser state is set to null.